truffle/contracts/VaultNotary/VaultNotary.sol

```diff
@@ -0,0 +1,233 @@
+
+    // Contract Events
+    event ContractDeprecatedSet(address indexed msgSender, bool isDeprecated);
+
+    /** @dev Modifier for limiting the access to vaultUri update
+      * only whitelisted user can do the decorated operation
+      */
+    modifier whitelistedOnlyForUri(bytes16 vaultId) {
+        //emit inside_whitelistedOnlyForUri(vaultId, msg.sender, msg.sender, notaryMapping[vaultId].vaultOwner);
+        //require(true);
+        require(msg.sender == notaryMapping[vaultId].vaultOwner || notaryMapping[vaultId].aclUriMapping[msg.sender],
+            "whitelistedOnlyForUri, only allow vault owner or the whitelisted users to access");
+        _;
+    }
+
+    /** @dev Modifier for limiting the access to vaultHash update
+      * only whitelisted user can do the decorated operation
+      */
+    modifier whitelistedOnlyForHash(bytes16 vaultId) {
+        require(msg.sender == notaryMapping[vaultId].vaultOwner || notaryMapping[vaultId].aclHashMapping[msg.sender],
+            "whitelistedOnlyForHash, only allow vault owner or the whitelisted users to access");
+        _;
+    }
+
+    /** @dev Modifier for limiting the access to grant and revoke permissions
+      * Will check the message sender, only the owner of a vault can do the operation decorated
+      * @param vaultId bytes16 ID of the vault to check
+      */
+    modifier vaultOwnerOnly(bytes16 vaultId) {
+        require(msg.sender == notaryMapping[vaultId].vaultOwner, "can only access by vault owner");
+        _;
+    }
+
+    /** @notice If we upgrade the version of the contract, the old contract
+      *  will not be allowed to register new Vault anymore. However, the existing
+      *  Vaults registered in that contract should still be able to be updated
+      */
+    modifier notDeprecated() {
+        require(!isDeprecated, "This version of the VaultNotary contract has been deprecated");
+        _;
+    }
+
+    /** @notice This is function to register a vault, will only do the registration if a vaultId is not
+                registered before
+      * @dev It sets the msg.sender to the vault owner and set the update permission of the owner to true
+      * It calls the setVaultUir and setVaultHash to initialize those two records
+      * @param vaultId bytes16 VaultID to create, is the same as shipment ID in our system
+      * @param vaultUri string Vault URI to set
+      * @param vaultHash string  Vault hash to set
+      */
+    function registerVault(bytes16 vaultId, string calldata vaultUri, string calldata vaultHash)
+        external
+        notDeprecated
+    {
+        require(isNotRegistered(vaultId), "vault should not be registered, in registerVault");
+        notaryMapping[vaultId].vaultOwner = msg.sender;
+
+        //work around for if (vaultUri != "")
+        bytes memory tempStringBytes = bytes(vaultUri);
+        if (tempStringBytes.length != 0)
```

Comment:
From what I'm reading, I think we can do this all in one line like
`bytes(vaultUri).length != 0`
which I would prefer instead of defining a variable.